packages/frontend/app/routes/peers.create.tsx

```diff
@@ -6,32 +6,43 @@ import {
   useNavigation
 } from '@remix-run/react'
 import { PageHeader } from '~/components/PageHeader'
-import { Button, ErrorPanel, Input } from '~/components/ui'
-import { type Asset } from '~/generated/graphql'
+import {
+  Button,
+  ErrorPanel,
+  Input,
+  Select,
+  type SelectOption
+} from '~/components/ui'
 import { listAssets } from '~/lib/api/asset.server'
 import { createPeer } from '~/lib/api/peer.server'
 import { commitSession, getSession, setMessage } from '~/lib/message.server'
 import { createPeerSchema } from '~/lib/validate.server'
 import type { ZodFieldErrors } from '~/shared/types'
 
 export async function loader() {
-  let assets: Asset[] = []
+  let assets: SelectOption[] = []
   let hasNextPage = true
   let after: string | undefined
 
   while (hasNextPage) {
-    const response = await listAssets({ after })
+    const response = await listAssets({ first: 100, after })
 
     if (response.edges) {
-      assets = [...assets, ...response.edges.map((edge) => edge.node)]
+      assets = [
+        ...assets,
+        ...response.edges.map((edge) => ({
+          label: `${edge.node.code} (Scale: ${edge.node.scale})`,
+          value: edge.node.id
+        }))
+      ]
```

Comment:
should this mapping be done outside `loader`? To decouple data-fetching from UI?